daemon/graphdriver/windows/windows.go

```diff
@@ -294,8 +294,10 @@ func (d *Driver) Remove(id string) error {
 		// not required.
 		computeSystems, err = hcsshim.GetContainers(hcsshim.ComputeSystemQuery{})
 		if err != nil {
-			if (osv.Build < 15139) &&
```

Comment:
15139 is somewhere between RS2 and RS3; from the context, it looks like the loop must be skipped on RS3 and up; I change it to an early return to make it slightly more readable

Reply:
Hmmm, no I think you've changed the logic incorrectly here.

Reply:
oh; why didn't I see this comment; I'll have to check

Reply:
Hm, I _think_ my logic is correct, but @lowenna @kolyshkin please double check;


(Simplified code, and replaced `15139 ` with `RS3` for brevity)


#### Old situation


```go
if err != nil {
	if (ver < RS3) && ((err ==InvalidState) || (err == AccessIsDenied)) {
		// retry
	}
	return err
}
```


ver       | error            | ver < RS3 ?        | `err == InvalidState or AccessIsDenied`? | action 
:---------|:-----------------|:-------------------|:-----------------------------------------|:-----------------
RS1       | nil              | -                  | -                                        | `break`
RS1       | any              | :white_check_mark: | :x:                                      | `return err`
RS1       | `InvalidState`   | :white_check_mark: | :white_check_mark:                       | retry
RS1       | `AccessIsDenied` | :white_check_mark: | :white_check_mark:                       | retry
RS3       | nil              | -                  | -                                        | `break`
RS3       | any              | :x:                | - (not evaluated)                        | `return err`
RS3       | `InvalidState`   | :x:                | - (not evaluated)                        | `return err`
RS3       | `AccessIsDenied` | :x:                | - (not evaluated)                        | `return err`

#### New situation:

```go
if err != nil {
	if ver >= RS3 {
		// no retry, return error
	}
	if (err == InvalidState) || (err == AccessIsDenied) {
		// retry
	}
	return err
}
```


ver       | error            | ver >= RS3 ?       | `err == InvalidState or AccessIsDenied`? | action 
:---------|:-----------------|:-------------------|:-----------------------------------------|:-----------------
RS1       | nil              | -                  | -                                        | `break`
RS1       | any              | :x:                | :x:                                      | `return err`
RS1       | `InvalidState`   | :x:                | :white_check_mark:                       | retry
RS1       | `AccessIsDenied` | :x:                | :white_check_mark:                       | retry
RS3       | nil              | -                  | -                                        | `break`
RS3       | any              | :white_check_mark: | - (not evaluated)                        | `return err`
RS3       | `InvalidState`   | :white_check_mark: | - (not evaluated)                        | `return err`
RS3       | `AccessIsDenied` | :white_check_mark: | - (not evaluated)                        | `return err`

Reply:
The new logic looks correct to me.

By the way, I found an obsoleted TODO comment just above this. I think the part of it needs to be removed, this one:

`For RS3, we can remove the retries. Also `